### ISSUE TITLE
PL/1 language server throws errors when current word matches code completion suggestion, and keyword alters code outline

### ISSUE THREAD

**Issue by jcnnrts** [IBM/zopeneditor-about#283]
## Development environment used

- Z Open Editor version:
- Editor Platform
  - [x] Visual Studio Code
  - [ ] Red Hat CodeReady Workspaces
  - [ ] Eclipse Che
  - [ ] Standalone Theia
- Editor Platform Version:
```
Version: 1.72.1 (user setup)
Commit: 129500ee4c8ab7263461ffe327268ba56b9f210d
Date: 2022-10-10T17:22:48.346Z
Electron: 19.0.17
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No
```

```
IBM Z Open Editor v2.2.0
Published - 9/14/2019, 01:42:42
Last released  - 9/19/2022, 23:12:23
Last updated - 9/20/2022, 21:18:33
Identifier - ibm.zopeneditor
```

- Operating System: Windows 10 Enterprise Edition
- Java Version (when using VS Code or Theia, execute `java -version` and paste the details here):
```
openjdk version "1.8.0_322"
OpenJDK Runtime Environment (Temurin)(build 1.8.0_322-b06)
OpenJDK 64-Bit Server VM (Temurin)(build 25.322-b06, mixed mode)
```
- [ ] Related to RSE API?
  - RSE API Plugin version:
  - Zowe CLI version:
  - Node.js version:
- Logs attached (see [here](https://ibm.github.io/zopeneditor-about/Docs/locating_local_client_logs.html) how to get them): no

## Problem Description

Detailed steps for reproducing the problem:

1. Start writing any PL/I program in a VSCode editor window where the PL/1 language server is active.
2. Manually finish entering a (key)word suggested by the code completion that would alter the code outline.

### Observed behavior

The PL/1 Language Server throws the following error on the last keystroke that completes a code completion suggestion:

```
[Error - 23:02:22] Request textDocument/foldingRange failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: java.lang.NullPointerException
	at java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:273)
	at java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:280)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:618)
	at java.util.concurrent.CompletableFuture$UniApply.tryFire(CompletableFuture.java:591)
	at java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:457)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1056)
	at java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1692)
	at java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:175)
Caused by: java.lang.NullPointerException
	at com.ibm.wazi.lsp.pli.core.handlers.FoldingRangeHandler.computeFoldingRanges(FoldingRangeHandler.java:42)
	at com.ibm.wazi.lsp.pli.core.PLILanguageServer.lambda$4(PLILanguageServer.java:199)
	at java.util.concurrent.CompletableFuture.uniApply(CompletableFuture.java:616)
	... 6 more
```

Example:

`
myProc: procedure;
`

The code completion will suggest "PROC","PROCEDURE","PROCEDURENAME","PROCNAME","PROD" when reaching

`
myproc: pro
`

The error will be thrown when manually completing "procedure" when the "c" in "proc" and the last "e" in "procedure" is typed. The error will not be thrown whenever you manually complete a suggestion if that suggestion is the only suggestion, such as "procedurename".

Furthermore, adding parameters to the procedure between parentheses will throw errors for every character typed between parentheses, presumably because the parameters alter the code outline.

### Expected behavior

No errors when manually completing code suggestions.

### Additional note

The language server throws this exact same error when typing between parentheses "()" that are syntactically incorrect. For example on a random new line where you would type "(throw error)", you would get the textDocument/foldingRange error for each character typed between these parentheses.

**Comment by phaumer** [comment]
Thanks for reporting this issue. We will track it as a duplicate to https://github.com/IBM/zopeneditor-about/issues/277

**Comment by phaumer** [comment]
Fixed in Z Open Editor 3.0.0
